gh-132054: Add ``application/yaml`` to ``mimetypes``

Kristinita opened the pull request:
<!-- gh-issue-number: gh-132054 -->
* Issue: gh-132054
<!-- /gh-issue-number -->

Kristinita committed:
gh-132054 Add `application/yaml` to `mimetypes`

AA-Turner commented on Misc/NEWS.d/next/Library/2025-04-03-20-28-54.gh-issue-132054.c1nlOx.rst:
```diff
@@ -0,0 +1 @@
+Mime type ``application/yaml`` is now supported by :mod:`mimetypes`.
```
```suggestion
The ``application/yaml`` mime type (:rfc:`9512`) is now supported
by :mod:`mimetypes`.
```

AA-Turner committed:
Update NEWS fragment

Kristinita committed:
fix(error): remove .yml from the list of preferred extensions

https://github.com/python/cpython/actions/runs/14249517987/job/39938739555?pr=132056#step:11:655

Kristinita committed:
fix(git): undo extra changes from the last push use “git restore” command

https://github.com/python/cpython/pull/132056#issuecomment-2776805064
Signed-off-by: Kristinita <Kristinita@users.noreply.github.com>

hugovk committed:
Test guess_file_type with types with multiple extensions

hugovk committed:
Omit ambiguous .mpa (audio/mpeg on Windows)

hugovk committed:
Add What's New entry

hugovk committed:
Update What's New entry

hugovk commented on Doc/whatsnew/3.14.rst:
```diff
@@ -829,6 +829,9 @@ mimetypes
   (Contributed by Hugo van Kemenade in :gh:`129965`.)
 
+* Add :rfc:`9512` ``application/yaml`` MIME type for YAML files (``.yaml``
+  and ``.yml``). (Contributed by Kristinita in :gh:`132056`.)
```
@Kristinita We can put your full name here if you like.

Kristinita replied:
@hugovk, thanks for adding the function `test_guess_file_type()`!

My contribution here is insignificant, and I think it would be more fair if you will add your name instead of my.

(But if the name of the author of the pull request is required, perhaps, `Sasha “Nelie” Chernykh` instead of `Kristinita` would be better).

Thanks.

hugovk replied:
You contribution is not insignificant. I've put both our names there, thanks again!

hugovk reviewed (approved):
Updated to add tests and a What's New entry.

The What's New for `mimetypes` is getting a bit sprawling, but we can copyedit that later as part of https://github.com/python/cpython/issues/123299.

@Kristinita Thank you and congratulations on your first contribution to CPython!

hugovk committed:
Update credit
